### ISSUE TITLE
GGVPointer does not honour custom raycast provider

### ISSUE THREAD

**Issue by keith-phillips** [microsoft/MixedRealityToolkit-Unity#11140]
## Overview
In our project we have registered a custom raycast provider in the input system, that has a number of special rules around what is considered a valid hit. After upgrading to MRTK 2.8, we now get unwanted focus events when using the GGV pointer, as it ignores our custom raycast provider and performs raycasts directly itself.

I assume the intent was not to prevent the use of custom raycast providers for the GGV pointer specifically, but it has certainly broken our use-case

## Expected behavior
Pointers should all use the custom raycast provider configured in the input system

## Actual behavior
Any pointers that implement 'IMixedRealityQueryablePointer' are ignoring the registered raycast provider

## Steps to reproduce
Don't have one  to share, sorry. It is obvious, however, from the method 'FocusProvider:QuerySceneWithPointer()', which accepts the raycast provider and doesn't use it.

## Unity editor version
2019.4

## Mixed Reality Toolkit release version
2.8.1

**Comment by davidkline-ms** [comment]
@keith-phillips, if there is a straight forward fix, please feel free to open a pr :smile:

**Comment by keith-phillips** [comment]
@davidkline-ms I feel like there should maybe an interface change to pass the registered IMixedRealityRaycastProvider through to the methods of IMixedRealityQueryablePointer, but I don't want to go making interface changes.

I've made the most minimal fix I can, which is to resolve and use the IMixedRealityRaycastProvider in the GGVPointer methods. I tried submitting a PR but messed it up a bit, I'll try again later if I can.

The ability to register a custom raycast provider is a hugely useful extension point in the MRTK, so it would be great to keep it functioning consistently.
